src/lib/proof-system/zkprogram.ts

```diff
@@ -527,9 +577,12 @@ function ZkProgram<
       rawMethods: Object.fromEntries(
         methodKeys.map((key) => [key, methods[key].method])
       ) as any,
+      proofsEnabled: doProving,
       setProofsEnabled(proofsEnabled: boolean) {
         doProving = proofsEnabled;
       },
+
+      proveRecursively,
```

Comment:
I will let the others be the judge if we want to express the API this way or if this is too confusing. I wonder if we can/should type this method in a certain way so its only available when its called in the correct context (within another program), and not in the plain native verison to avoid confusion

Reply:
one note: `proveRecursively` doesn't return you a proof (just the proof output), and it's therefore hard to confuse with the plain proving API.

like, if you need a proof, you'll figure out quickly that this API doesn't give you what you want.

(if you don't care about a proof, then it's fine to use this API outside a circuit, because it does run the method)

Reply:
But for the example, I wouldn't understand how to use this.  `.recursiveProvers` seems slightly better, at least because it sounds like it may return an object.  It still sounds like `.recursiveProvers.baseCase()` will return a proof, so potentially `.recursiveProverFunctions` is the most natural sounding.

I think there's an asterisk to this which is that zkprogram is still a very opaque API that people don't have an intuitive sense for, so I think we can only be so intuitive with the naming anyways.